Assets/MixedRealityToolkit/_Core/Definitions/Utilities/InscribedRectangle.cs

```diff
@@ -0,0 +1,570 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities
+{
+    /// <summary>
+    /// The InscribedRectangle class defines the largest rectangle within an
+    /// arbitrary shape.
+    /// </summary>
+    public class InscribedRectangle
+    {
+        /// <summary>
+        /// Total number of starting points randomly generated within the boundary.
+        /// </summary>
+        private static readonly int randomPointCount = 30;
+
+        /// <summary>
+        /// The total amount of height, in meters,  we want to gain with each binary search
+        /// change before we decide that it's good enough.
+        /// </summary>
+        private const float minimumHeightGain = 0.01f;
+
+        /// <summary>
+        /// The center point of the inscribed rectangle.
+        /// </summary>
+        public Vector2 Center { get; private set; } = EdgeUtilities.InvalidPoint;
+
+        /// <summary>
+        /// The width of the inscribed rectangle.
+        /// </summary>
+        public float Width { get; private set; } = 0f;
+
+        /// <summary>
+        /// The height of the inscribed rectangle.
+        /// </summary>
+        public float Height { get; private set; } = 0f;
+
+        /// <summary>
+        /// The rotation angle, in degrees, of the inscribed rectangle.
+        /// </summary>
+        public float Angle { get; private set; } = 0f;
+
+        /// <summary>
+        /// Is the described rectangle valid?
+        /// </summary>
+        public bool IsValid
+        {
+            get
+            {
+                // A rectangle is considered valid if it's center point is valid.
+                return EdgeUtilities.IsValidPoint(Center);
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="geometryEdges"></param>
+        /// <param name="randomSeed"></param>
+        public InscribedRectangle(Edge[] geometryEdges, int randomSeed)
+        {
+            FindInscribedRectangle(geometryEdges, randomSeed);
```

Comment:
Why not just make `FindInscribedRectangle` public?

Reply:
this is how the original code was written. you generally arent going to do it more than once per instance of the app (boundaries are system setup thing)

Reply:
Let's get rid of the redundant public method and just rename the private one to match.